### ISSUE TITLE
[ORM] Relations to the same model with composite keys not working properly

### ISSUE THREAD

**Issue by marianacapelo** [phalcon/cphalcon#10368]
Hi. I'm using Models in my project, and I have been struggling with one that references itself - a "parent" row. But this table has a composite key, which envolves an id and a language id. I defined the 2 relations as follows:

```php
$this->hasMany(array('id','language_id'),'Libraries\Models\Pages', array('parent_id','parent_language_id'),array('alias'=>'children'));

$this->belongsTo(array('parent_id','parent_language_id'),'Libraries\Models\Pages', array('id','language_id'),array('alias'=>'parent'));
```

However this does not work correctly. I have been over this multiple times and can't seem to find an error on my part. Did I miss something or is it actually a bug?

**Comment by andresgutierrez** [comment]
Which part is not working for you? Are you getting an error or something?

**Comment by marianacapelo** [comment]
I don't get the right results using the relations defined.
For example, with the parent relation, if I try to fetch a row with id = 101, language id = 1 and parent_id = 100 and parent_language_id = 1:

```
   $page = Pages::findFirst('id=101 AND language_id=1'); 
   echo '<br>From parent columns of entry: '.$page->parent_id.' '.$page->parent_language_id;
   echo '<br>From parent relation:'.$page->parent->id.' '.$page->parent->language_id;
```
The data I get from the parent relation is not right. The output is the following:

```
From parent columns of entry: 100 1
From parent relation:101 1
```

Don't know if makes a difference, but the model has the property $id and $language_id both as primary.

**Comment by marianacapelo** [comment]
And, in reverse, if I try to access the children from the entry with id 100 and language id = 1 (which should include the entry with id=101), I get all entries that share the same parent_id as the entry with id=100 (instead of the entries with parent_id=100). 
If I remove the composite key in the relation definition, I can access the children just fine.

**Comment by marianacapelo** [comment]
@andresgutierrez do you know if I'm the first to find a situation like this?

**Comment by andresgutierrez** [comment]
I will need some time to reproduce the issue

**Comment by sergeyklay** [comment]
Fixed in the `3.0.x` branch.
